src/Drupal/Composer/DrupalInstallerPlugin.php

```diff
@@ -520,6 +518,10 @@ protected function afterAllPatchesGitBranchCleanup(PackageInterface $package) {
         $this->verifyGitBranchExists($this->git['base-branch']);
 
         $isGitDiff = $this->isGitDiff($branchName);
+        if ($isGitDiff) {
+          $this->afterCommit($package);
+        }
```

Comment:
This slipped in, but makes generally more sense to check if the branch is identical to the base branch.